packages/SwingSet/test/upgrade-swingset.test.js

```diff
@@ -229,3 +255,196 @@ test('upgrade non-reaping kernel state', async t => {
+
+  // fourth promise has additional references, still remediated
+  kvStore.set(`${p4}.state`, 'rejected');
+  kvStore.set(`${p4}.data.body`, dccd.body);
+  kvStore.set(`${p4}.data.slots`, '');
+  // note: we aren't being specific about *where* the other reference
+  // is coming from. A plausible source is an argument of a message
+  // queued to some other unresolved promise. A non-plausible one is
+  // in the c-list of some other vat (as a settled promise that one
+  // should have gotten a notify too, assuming they were subscribed,
+  // and they shouldn't be not subscribed). If the refcounts were
+  // stored in a DB with more runtime checking, we'd be creating an
+  // illegal situation here, but it's not.
+  kvStore.set(`${p4}.refCount`, '2'); // c-list, other
+  kvStore.set(`${vatID}.c.${p4}`, 'R p+93');
+  kvStore.set(`${vatID}.c.p+93`, p4);
+
+  // fifth promise is fulfilled, not rejected, without a notify:
+  // remediated (even though strictly speaking 9039 is about rejected
+  // promises)
+  kvStore.set(`${p5}.state`, 'fulfilled');
+  kvStore.set(`${p5}.data.body`, '#{}');
+  kvStore.set(`${p5}.data.slots`, '');
+  kvStore.set(`${p5}.refCount`, '1');
+  kvStore.set(`${vatID}.c.${p5}`, 'R p+95');
+  kvStore.set(`${vatID}.c.p+95`, p5);
+
+  // sixth promise is unresolved: ignored
+  kvStore.set(`${p6}.state`, 'unresolved');
+  kvStore.set(`${p6}.subscribers`, '');
+  kvStore.set(`${p6}.queue.nextID`, `0`);
+  kvStore.set(`${p6}.refCount`, `1`);
+  kvStore.set(`${p6}.decider`, vatID);
+  kvStore.set(`${vatID}.c.${p6}`, 'R p+96');
+  kvStore.set(`${vatID}.c.p+96`, p6);
+
+  // now update queues
+
+  // eslint-disable-next-line prefer-const
+  let [runHead, runTail] = JSON.parse(kvStore.get('runQueue'));
+  for (const m of runQueue) {
+    kvStore.set(`runQueue.${runTail}`, JSON.stringify(m));
+    runTail += 1;
+  }
+  kvStore.set('runQueue', JSON.stringify([runHead, runTail]));
+
+  // eslint-disable-next-line prefer-const
+  let [accHead, accTail] = JSON.parse(kvStore.get('acceptanceQueue'));
+  for (const m of acceptanceQueue) {
+    kvStore.set(`acceptanceQueue.${accTail}`, JSON.stringify(m));
+    accTail += 1;
+  }
+  kvStore.set('acceptanceQueue', JSON.stringify([accHead, accTail]));
+
+  const stats = JSON.parse(kvStore.get('kernelStats'));
+  stats.runQueueLength += runQueue.length;
+  stats.runQueueLengthUp += runQueue.length;
+  stats.runQueueLengthMax = runQueue.length;
+  stats.acceptanceQueueLength += acceptanceQueue.length;
+  stats.acceptanceQueueLengthUp += acceptanceQueue.length;
+  stats.acceptanceQueueLengthMax = acceptanceQueue.length;
+  kvStore.set('kernelStats', JSON.stringify(stats));
+
+  await commit();
+
+  const data = { ...debug.dump().kvEntries };
+
+  // confirm that this state is too old for the kernel to use
+  await t.throwsAsync(() => makeSwingsetController(kernelStorage), {
+    message: /kernel DB is too old/,
+  });
+
+  // upgrade it
+  upgradeSwingset(kernelStorage);
```

Comment:
Check that `modified` is true ?

```suggestion
  const { modified } = upgradeSwingset(kernelStorage);
  t.true(modified);
```